### PR TITLE
In payment

### PR DESCRIPTION
**Pull request by samsoncsyu7777**
### What this PR does (required):
- createPaymentIntent service to create payment intent in Stripe to confirm the payment
- create @route POST /:id/checkout-with-created-intent route to get Customer id or create a new Customer from Stripe before confirming the payment
- create confirmPayments API call in FE to pass three parameters to /:id/checkout-with-created-intent: totalCost, default payment method, and user id
- create a confirm payment button with totalCost as the parameter and gather payment method and user id from profile context and user context

### Screenshots / Videos (front-end only):
Screenshot 2021-11-25 at 9 38 08 PM

### Any information needed to test this feature (required):
The button will be integrated into request page and tested later.

### CODE REVIEW

File: client/src/components/EditProfile/EditPhoto/EditPhoto.tsx [hunk 1]
```diff
@@ -1,77 +1,93 @@
...
+          </Box>
+          <Box display="flex" alignItems="center" justifyContent="center">
+            <Typography className={classes.reminder} component="h4" variant="h6">
+              Be sure to use a photo that clearly shows your face
+            </Typography>
+          </Box>
+          <Box className={classes.centerRow}>
+            <Button component="label" className={classes.uploadButton} variant="outlined" color="primary">
+              Upload a file from your device
+              <input
```
Comment (sundayezeilo) [inline client/src/components/EditProfile/EditPhoto/EditPhoto.tsx]: Please always use MUI components whenever possible. Use a `TextField` component for this one.

File: server/controllers/payment.js [hunk 2]
```diff
@@ -1,65 +1,211 @@
...
+
+    res
+      .status(200)
+      .send({ message: "The customer has been successfully created", customer: req.createdCustomer });
+  }
+});
+
+// @route GET /:id/new-customer-create-intent
+// @desc create customer and intent
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js]: Good documentation

File: server/controllers/payment.js [hunk 3]
```diff
@@ -1,65 +1,211 @@
...
+        message: "Both customer and intent have been successfully created",
+        customer: req.createdCustomer,
+        intent: req.intent
+      });
+  }
+});
+
+// @route GET /:id/add-card
+// @desc get or create customer and create intent
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js]: :+1:

File: server/controllers/payment.js [hunk 4]
```diff
@@ -1,65 +1,211 @@
...
+        message: "Both customer and intent have been successfully created",
+        customer: req.createdCustomer,
+        intent: req.intent
+      });
+  }
+});
+
+// @route POST /:id/checkout-with-created-intent
+// @desc create payment intent with totalCost and paymentMethod
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js]: :+1:

File: server/controllers/payment.js [hunk 5]
```diff
@@ -1,65 +1,211 @@
...
+        message: "Both customer and paymentIntent have been successfully created",
+        customer: req.createdCustomer,
+        paymentIntent: req.paymentIntent
+      });
+  }
+});
+
+// @route GET /:id/all-payment-methods
+// @desc get all payment methods
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js]: :+1:

File: server/controllers/payment.js [hunk 6]
```diff
@@ -1,65 +1,211 @@
...
+        .send({
+          message: "New customer without records"
+        });
+    }
+  }
+});
+
+// @route GET /:id/add-card
+// @desc get or create customer and create intent
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js]: Nice documentation

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Remarkable work on this one, though it took a lot all those changes :smiley: 
Please see the review comments and address them. 
Happy coding!

File: client/src/components/EditProfile/EditPhoto/EditPhoto.tsx (hunk 1 above)
Comment (samsoncsyu7777) [reply client/src/components/EditProfile/EditPhoto/EditPhoto.tsx]: There are some features needed that are not supported by TextField.

File: server/controllers/payment.js (hunk 2 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js]: Thanks.

File: server/controllers/payment.js (hunk 3 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js]: Thanks.

File: server/controllers/payment.js (hunk 5 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js]: Thanks.

File: server/controllers/payment.js (hunk 6 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js]: Thanks.

File: client/src/components/EditProfile/EditPhoto/EditPhoto.tsx (hunk 1 above)
Comment (sundayezeilo) [reply client/src/components/EditProfile/EditPhoto/EditPhoto.tsx]: > There are some features needed that are not supported by TextField.

That's true for file upload. :+1:

File: server/controllers/payment.js (hunk 4 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js]: Thanks.

**Review by rajivtitus — APPROVED** [review]
Impressive work with the entire payment feature 💯

**Pull request merged**